src/libraries/System.Security.Cryptography.X509Certificates/tests/CertTests.cs

```diff
@@ -495,6 +496,15 @@ public static void MutateDistinguishedName_SubjectName_DoesNotImpactSubject()
             }
         }
 
+        [Fact]
+        public static void SignatureAlgorithmOidReadableForUnknownKeyAlgorithms()
```

Comment:
In order to be truly unknown we'd probably need to build up a cert using a nonsense OID.  But since most Windows machines don't know about the GOST algorithms this is sufficient for now.  (Because GOST is really the scenario here, the only change I'd make at this point would be to rename the method)

For a theoretical future test, the best nonsense OID would use the open-GUID space, but Win32 doesn't support reading those, so it's "the best, but not an option".